### ISSUE TITLE
Selecting bin count on intermediate data question fails

### ISSUE THREAD

**Issue by KennethLim314** [metabase/metabase#16379]
**Describe the bug**
When using an existing question as the intermediate data, grouping by a continuous value and selecting the bin count returns an error:
```
Value does not match schema: {:query {:breakout [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}
```
The error does not occur for auto-bin

**Logs**
Please include javascript console and server logs around the time this bug occurred. For information about how to get these, consult our [bug troubleshooting guide](https://metabase.com/docs/latest/troubleshooting-guide/bugs.html)
```
[4e41f61c-4ae4-419e-ae00-dbb2bfd8b312] 2021-06-07T11:35:57+08:00 INFO metabase.api.dataset Source query for this query is Card 206
[4e41f61c-4ae4-419e-ae00-dbb2bfd8b312] 2021-06-07T11:35:57+08:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 1,
 :started_at #t "2021-06-07T03:35:57.330159Z[GMT]",
 :error_type :invalid-query,
 :json_query
 {:database 1,
  :query
  {:source-table "card__206",
   :aggregation [["distinct" ["field" 3 nil]]],
   :breakout [["field" "avg" {:binning {:strategy "num-bins", :num-bins 50}}]]},
  :type "query",
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> util.schema$schema_core_validator$fn__17906.invoke(schema.clj:29)"
  "query_processor.middleware.validate$validate_query$fn__50505.invoke(validate.clj:9)"
  "query_processor.middleware.normalize_query$normalize$fn__48386.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__45652.invoke(add_rows_truncated.clj:35)"
  "metabase_enterprise.audit.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__31908.invoke(handle_audit_queries.clj:162)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__50436.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__47503.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__49548.invoke(process_userland_query.clj:135)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__47443.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___33764$thunk__33765.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___33764.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___33773$fn__33776.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___33773.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:241)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:237)"
  "query_processor$fn__51113$process_query_and_save_execution_BANG___51122$fn__51125.invoke(query_processor.clj:253)"
  "query_processor$fn__51113$process_query_and_save_execution_BANG___51122.invoke(query_processor.clj:245)"
  "query_processor$fn__51157$process_query_and_save_with_max_results_constraints_BANG___51166$fn__51169.invoke(query_processor.clj:265)"
  "query_processor$fn__51157$process_query_and_save_with_max_results_constraints_BANG___51166.invoke(query_processor.clj:258)"
  "api.dataset$run_query_async$fn__66371.invoke(dataset.clj:56)"
  "query_processor.streaming$streaming_response_STAR_$fn__66350$fn__66351.invoke(streaming.clj:72)"
  "query_processor.streaming$streaming_response_STAR_$fn__66350.invoke(streaming.clj:71)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__17604.invoke(streaming_response.clj:84)"],
 :context :ad-hoc,
 :error
 "Value does not match schema: {:query {:breakout [(named (not (\":field clauses using a string field name must specify :base-type.\" a-clojure.lang.PersistentVector)) \"Must be a valid instance of one of these clauses: :expression, :field\")]}}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :schema.core/error,
  :value
  {:database 1,
   :query
   {:source-table "card__206",
    :aggregation [[:distinct [:field 3 nil]]],
    :breakout [[:field "avg" {:binning {:strategy :num-bins, :num-bins 50}}]]},
   :type :query,
   :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
   :info
   {:executed-by 1,
    :context :ad-hoc,
    :card-id 206,
    :nested? true,
    :query-hash
    [7, 5, 85, -45, 30, 123, 48, 8, 68, -98, -109, 62, -44, 124, -72, 53, 91, -48, -29, 120, -91, -98, -56, 17, 30,
     -50, -54, -68, -31, -60, -30, -71]},
   :constraints {:max-results 10000, :max-results-bare-rows 2000}},
  :error
  {:query
   {:breakout
    [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}},
 :data {:rows [], :cols []}}
```

**To Reproduce**
Using orders table, create an intermediate table as follows:
AVERAGE OF Subtotal, GROUP BY user_id.

Then, create a second question using the first question as the source data.
Summarize as follows:
COUNT, GROUP BY average of SUBTOTAL
Then, select the number of bins. This should raise an error

**Expected behavior**
We get a histogram with a fixed number of bins

**Screenshots**
![image](https://user-images.githubusercontent.com/47406716/120956390-06e2bd00-c786-11eb-903e-2284b703e0a6.png)

![image](https://user-images.githubusercontent.com/47406716/120956355-f6cadd80-c785-11eb-9fcb-7cb30b9d1b4d.png)

**Information about your Metabase Installation:**

```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.11+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.11+9",
    "os.name": "Linux",
    "os.version": "4.14.219-164.354.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "bigquery",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.9"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-05-27",
      "tag": "v1.39.3",
      "branch": "release-x.39.x",
      "hash": "598a112"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Annoying, breaks a few questions in our dashboards

**Additional context**

**Comment by flamber** [comment]
Very related to #15446/#16327, but this is specific to number bins. Let's make sure that this is fixed for both GUI and SQL questions used for Saved Questions.

**Comment by ariya** [comment]
The reason the query threw an error was because the missing `base-type`. This is apparently also fixed by PR #16707.

Before PR #16707, this is how it looked like:


```json
{
   "type":"query",
   "query":{
      "source-table":"card__1",
      "aggregation":[
         [
            "count"
         ]
      ],
      "breakout":[
         [
            "field",
            "avg",
            {
               "base-type":"type/Float",
               "binning":{
                  "strategy":"num-bins",
                  "num-bins":10
               }
            }
         ]
      ]
   },
   "database":1,
   "parameters":[]
}
```

Strangely, automatic binning still specified the `base-type`, hence why it somehow works:

```json
{
   "type":"query",
   "query":{
      "source-table":"card__6",
      "aggregation":[
         [
            "count"
         ]
      ],
      "breakout":[
         [
            "field",
            "avg",
            {
               "binning":{
                  "strategy":"default"
               },
               "base-type":"type/Float"
            }
         ]
      ]
   },
   "database":1,
   "parameters":[]
}
```

After PR #16707, the breakout in the query always contains `base-type`, restoring the behavior as in v38 and earlier, and that solves the issue:


```json
{
   "type":"query",
   "query":{
      "source-table":"card__1",
      "aggregation":[
         [
            "count"
         ]
      ],
      "breakout":[
         [
            "field",
            "avg",
            {
               "base-type":"type/Float",
               "binning":{
                  "strategy":"num-bins",
                  "num-bins":10
               }
            }
         ]
      ]
   },
   "database":1,
   "parameters":[]
}
```
